src/storage/store/node_table.cpp

```diff
@@ -68,46 +88,33 @@ void NodeTable::initializePKIndex(const std::string& databasePath,
         *memoryManager->getBufferManager(), shadowFile, vfs, context);
 }
 
-void NodeTable::initializeScanState(Transaction* transaction, TableScanState& scanState) {
+void NodeTable::initScanState(Transaction* transaction, TableScanState& scanState) {
     auto& nodeScanState = scanState.cast<NodeTableScanState>();
+    NodeGroup* nodeGroup = nullptr;
     switch (nodeScanState.source) {
     case TableScanSource::COMMITTED: {
-        nodeScanState.nodeGroup = nodeGroups->getNodeGroup(nodeScanState.nodeGroupIdx);
+        nodeGroup = nodeGroups->getNodeGroup(nodeScanState.nodeGroupIdx);
     } break;
     case TableScanSource::UNCOMMITTED: {
         const auto localTable = transaction->getLocalStorage()->getLocalTable(tableID,
             LocalStorage::NotExistAction::RETURN_NULL);
         KU_ASSERT(localTable);
         auto& localNodeTable = localTable->cast<LocalNodeTable>();
-        nodeScanState.nodeGroup = localNodeTable.getNodeGroup(nodeScanState.nodeGroupIdx);
-        KU_ASSERT(nodeScanState.nodeGroup);
+        nodeGroup = localNodeTable.getNodeGroup(nodeScanState.nodeGroupIdx);
+        KU_ASSERT(nodeGroup);
     } break;
-    default: {
+    case TableScanSource::NONE: {
         // DO NOTHING.
+    } break;
+    default: {
+        KU_UNREACHABLE;
```

Comment:
LCOV_EXCL...

Reply:
Added a wrap of lcov excl in `KU_UNREACHABLE`, let's see if that can solve the problem globally.